simulation/traffic_simulator/include/traffic_simulator/utils/node_parameter_handler.hpp

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TRAFFIC_SIMULATOR__UTILS__NODE_PARAMETER_HANDLER_HPP_
+#define TRAFFIC_SIMULATOR__UTILS__NODE_PARAMETER_HANDLER_HPP_
+
+#include <rclcpp/rclcpp.hpp>
+#include <scenario_simulator_exception/exception.hpp>
+
+namespace traffic_simulator
+{
+class NodeParameterHandler
```

Comment:
Is this class necessary?
I feel that the NodeParameterHandler::getParameter function should also be implemented as a free function that takes node_parameters_interface_ as an argument.

Reply:
This class is not strictly necessary. I have implemented the functionality as a class, because when this simple class was passed around it was a nice wrapper over `NodeParametersInterface::SharedPtr` that provided an exact functionality that was needed. When the function received the class pointer as argument, it was clear what this argument was for. Furthermore, if some additional functionality was to be developed in the future, like for example parameter caching, this class would be the place to do this.

I have made the changes you requested and changed the `getParameter` to be a free function in [this commit](https://github.com/tier4/scenario_simulator_v2/pull/1275/commits/2a0cb5aa63e5a7c6a8a9de5e389d8631dfd3f2aa).
If you think the class solution was better, I can revert this commit.